### PR TITLE
change raise style in msct_parser

### PR DESCRIPTION
**Pull request by poquirion**
Implement @peristeri request in: https://github.com/neuropoly/spinalcordtoolbox/pull/1587#pullrequestreview-91677625

Using syntax error an value error in msct_parser. Try to tweak message to be a bit more explicit

### CODE REVIEW

File: scripts/msct_parser.py [hunk 1]
```diff
@@ -246,9 +246,9 @@ def checkFolderCreation(self, param):
         else:
             result_creation = 0  # no need for checking
         if result_creation == 2:
-            sct.printv("ERROR: Permission denied for folder creation...", type="error")
+            raise OSError("Permission denied for folder creation {}".format(param))
         elif result_creation == 1:
-            sct.printv("Folder " + param + " has been created.", 0, type='warning')
+            sct.log.warning("Folder " + param + " has been created.", 0)
```
Comment (peristeri) [inline scripts/msct_parser.py]: Creating a folder does not need to be a warning. printv without the warning should be fine. But due to the legacy code throwing a warning, I'm not sure what the motivation is.

Comment (poquirion) [reply scripts/msct_parser.py]: lets do `log.info`...

**Comment by poquirion** [comment]
Ok, I know what happens there

**Comment by poquirion** [comment]
@peristeri Is it better like this? Not sure I covered everything...

**Comment by perone** [comment]
I think a better approach would be to extend the `Exception` base class as recommended by the Python [official docs](https://docs.python.org/2/library/exceptions.html).
`SyntaxError` for instance is an exception that is raised by the Python parser for code, using it in user-code for invalid arguments would be a repurpose of the exception. As an alternative, if you don't want to derive custom classes from the `Exception` class, you can use `RuntimeError` for all fatal errors.

**Comment by jcohenadad** [comment]
@zougloub any comment?

**Comment by zougloub** [comment]
Also not a big fan of SyntaxError, but the rest (ValueError / OSError) looks OK to me.
I briefly tested the branch with 2 different kinds of mistaeks:
```python
.../sct_testing_data/t2 $ sct_label_vertebrae -i t2dqsd.nii.gz -s t2_seg.nii.gz -r 1 -c t2
... $ sct_label_vertebrae -i t2.nii.gz -s t2_seg.nii.gz -r 1 -c t2 -qsd
```

I notice that the output doesn't have the same style (first: no color, second: red and something like `sct_utils.pyNone`) (yes there was a similar, but not same, behaviour before), can this be improved at the same time?

File: scripts/msct_parser.py [hunk 2]
```diff
@@ -246,9 +246,9 @@ def checkFolderCreation(self, param):
         else:
             result_creation = 0  # no need for checking
         if result_creation == 2:
-            sct.printv("ERROR: Permission denied for folder creation...", type="error")
+            raise OSError("Permission denied for folder creation {}".format(param))
```
[the remark this answers is not in the document]
Comment (peristeri) [reply scripts/msct_parser.py]: I agree with you. I could do it for this file, but changing the entire code base out of the scope of this pull request.

[the remark this answers is not in the document]
Comment (perone) [reply scripts/msct_parser.py]: Sure, I opened another issue because of that, to avoid mixing these changes with this pull request.

**Pull request merged**